src/modules/75-cf/components/FlagArchiving/useRestoreFlagDialog.tsx

```diff
@@ -0,0 +1,59 @@
+import React from 'react'
+import { Intent } from '@harness/design-system'
+import { UseConfirmationDialogReturn, useToaster } from '@harness/uicore'
+import { String, useStrings } from 'framework/strings'
+import { useRestoreFeatureFlag, RestoreFeatureFlagQueryParams, Feature } from 'services/cf'
+import useResponseError from '@cf/hooks/useResponseError'
+import { useConfirmAction } from '@common/hooks'
+
+export interface RestoreFlagDialogProps {
+  flagData: Feature
+  queryParams: RestoreFeatureFlagQueryParams
+  refetchFlags?: () => void
+}
+
+const useRestoreFlagDialog = ({
+  flagData,
+  queryParams: restoreQueryParams,
+  refetchFlags
+}: RestoreFlagDialogProps): UseConfirmationDialogReturn['openDialog'] => {
+  const { showSuccess, showError } = useToaster()
+  const { getString } = useStrings()
+  const { handleResponseError } = useResponseError()
+
+  const queryParams = {
+    identifier: flagData.identifier,
+    environmentIdentifier: flagData.envProperties?.environment,
+    ...restoreQueryParams
+  }
+
+  const { mutate: restoreFeatureFlag } = useRestoreFeatureFlag({
+    identifier: flagData.identifier,
+    queryParams
+  })
+
+  const handleRestore = async (): Promise<void> => {
+    try {
+      await restoreFeatureFlag()
+      showSuccess(getString('cf.featureFlags.archiving.restoreSuccess'))
+      refetchFlags?.()
+    } catch (e) {
+      handleResponseError(e)
+      showError(e.message)
+    }
+  }
+
+  return useConfirmAction({
+    title: getString('cf.featureFlags.archiving.restoreFlag'),
+    intent: Intent.DANGER,
+    confirmText: getString('cf.featureFlags.archiving.restore'),
+    message: (
+      <String stringID="cf.featureFlags.archiving.restoreDescription" vars={{ flagName: flagData.name }} useRichText />
+    ),
```

Comment:
I think the `message` in the confirmation dialog has a bit of stupid CSS attached to get text to break if it's too long, but we poorly done and causes words to get cut off in weird ways. We might need to add a wrapper around the content and apply the correct CSS.
```suggestion
    message: (
      <span style={{ wordBreak: 'break-word' }}>
        {getString('cf.featureFlags.archiving.restoreDescription', { flagName: flagData.name })}
      </span>
    ),
```